QueryEngine/ResultSetConversion.cpp

```diff
@@ -518,6 +526,65 @@ arrow::RecordBatch ResultSet::convertToArrow(const std::vector<std::string>& col
   return arrow::RecordBatch(schema, row_count, columns);
 }
 
+// WARN(ptaylor): users are responsible for detaching and removing shared memory segments, e.g.,
+//   int shmid = shmget(...);
+//   char *ipc_ptr = (char* shmat(shmid, ...));
+//   ...
+//   shmdt(ipc_ptr);
+//   shmctl(shmid, IPC_RMID, 0);
+std::tuple<std::shared_ptr<arrow::Buffer>, std::vector<char>, int64_t> ResultSet::getArrowCopy(
```

Comment:
@trxcllnt you can leave this change to return type to me, we are coming up with new thrift API to merge CPU and GPU path.